CMakeLists.txt

```diff
@@ -0,0 +1,91 @@
+# Copyright (c) 2023 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+cmake_minimum_required(VERSION 3.13)
+
+if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.15)
+  # MSVC runtime library flags are selected by the CMAKE_MSVC_RUNTIME_LIBRARY abstraction.
+  cmake_policy(SET CMP0091 NEW)
+  # MSVC warning flags are not in CMAKE_<LANG>_FLAGS by default.
+  cmake_policy(SET CMP0092 NEW)
+endif()
+if(CMAKE_VERSION VERSION_GREATER_EQUAL 3.20)
+  # MSVC RTTI flag /GR is not added to CMAKE_CXX_FLAGS by default.
+  cmake_policy(SET CMP0117 NEW)
+endif()
+
+project("Bitcoin Core"
+  VERSION 24.99.0
+  DESCRIPTION "Bitcoin client software"
+  LANGUAGES CXX C ASM
+)
+
+set(PACKAGE_NAME ${PROJECT_NAME})
+set(PACKAGE_VERSION ${PROJECT_VERSION})
+set(CLIENT_VERSION_MAJOR ${PROJECT_VERSION_MAJOR})
+set(CLIENT_VERSION_MINOR ${PROJECT_VERSION_MINOR})
+set(CLIENT_VERSION_BUILD ${PROJECT_VERSION_PATCH})
+set(CLIENT_VERSION_IS_RELEASE "false")
+set(PACKAGE_BUGREPORT "https://github.com/bitcoin/bitcoin/issues")
+set(PACKAGE_URL "https://bitcoincore.org/")
+set(COPYRIGHT_YEAR "2023")
+set(COPYRIGHT_HOLDERS "The %s developers")
+set(COPYRIGHT_HOLDERS_SUBSTITUTION ${PROJECT_NAME})
+set(COPYRIGHT_HOLDERS_FINAL "The Bitcoin Core developers")
+
+# Configurable options.
+# When adding a new option, end the <help_text> with a full stop for consistency.
+include(CMakeDependentOption)
+cmake_dependent_option(CXX20 "Enable compilation in C++20 mode." OFF "NOT MSVC" ON)
+
+if(CXX20)
+  set(CMAKE_CXX_STANDARD 20)
```

Comment:
I changed this to 26 for testing, but didn't run into an issue on cmake 3.13. Is this expected?

Reply:
> I changed this to 26 for testing, but didn't run into an issue on cmake 3.13. Is this expected?

Mind providing the exact command?

Reply:
```
# grep CMAKE_CXX_STANDARD CMakeLists.txt && cmake --version && cmake   -S . -B ./build7  
  set(CMAKE_CXX_STANDARD 26)
  set(CMAKE_CXX_STANDARD 26)
set(CMAKE_CXX_STANDARD_REQUIRED ON)
cmake version 3.13.4

CMake suite maintained and supported by Kitware (kitware.com/cmake).
-- The CXX compiler identification is GNU 8.3.0
-- The C compiler identification is GNU 8.3.0
-- The ASM compiler identification is GNU
-- Found assembler: /usr/bin/cc
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done


Configure summary
=================
Preprocessor defined macros ........... 
C compiler ............................ /usr/bin/cc
CFLAGS ................................ 
C++ compiler .......................... /usr/bin/c++
CXXFLAGS .............................. 
Common compile options ................ 
Common link options ................... 
Build type:
 - CMAKE_BUILD_TYPE ................... 
 - CFLAGS ............................. 
 - CXXFLAGS ........................... 
 - LDFLAGS for executables ............ 
 - LDFLAGS for shared libraries ....... 


-- Configuring done
-- Generating done
-- Build files have been written to: /bitcoin-core/build7

Reply:
The [`CMAKE_CXX_STANDARD`](https://cmake.org/cmake/help/latest/variable/CMAKE_CXX_STANDARD.html) variable just holds a default value for [`CXX_STANDARD`](https://cmake.org/cmake/help/latest/prop_tgt/CXX_STANDARD.html) target property, which CMake will actually check.

To see such behavior, please consider applying a diff as follows:
```diff
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -89,3 +89,5 @@ else()
   message(" - LDFLAGS for shared libraries ....... ${CMAKE_SHARED_LINKER_FLAGS_RELEASE}")
 endif()
 message("\n")
+
+add_library(test_lib OBJECT src/addrdb.cpp)
```